# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1295428226 18000
#      Wed Jan 19 04:10:26 2011 -0500
# Node ID 90f8d12f196497d941403a69e99463fd049d2d9a
# Parent  3c6e8aaa9555ac682fd959f3b8eb18f8247be901
avoid PCRE-ism in regexp

diff --git a/scripts/ChangeLog b/scripts/ChangeLog
--- a/scripts/ChangeLog
+++ b/scripts/ChangeLog
@@ -1,8 +1,12 @@
+2011-01-19  John W. Eaton  <jwe@octave.org>
+
+	* io/strread.m: Avoid PCRE-ism in regexp.  Bug #32066.
+
 2011-01-17  Rik  <octave@nomad.inbox5.com>
 
 	* audio/wavread.m, deprecated/spqr.m, elfun/lcm.m, general/bitget.m,
 	general/curl.m, general/divergence.m, general/interp3.m,
 	general/quadv.m, geometry/delaunayn.m, geometry/griddata.m,
 	help/help.m, io/textread.m, io/textscan.m, linear-algebra/cond.m,
 	miscellaneous/mkoctfile.m, miscellaneous/orderfields.m,
 	miscellaneous/run.m, miscellaneous/setappdata.m, plot/axis.m,
diff --git a/scripts/io/strread.m b/scripts/io/strread.m
--- a/scripts/io/strread.m
+++ b/scripts/io/strread.m
@@ -183,17 +183,17 @@ function varargout = strread (str, forma
     len = length (str);
     c2len = length (comment_specif{2});
     str = cellslices (str, [1, cstop + c2len], [cstart - 1, len]);
     str = [str{:}];
   endif
 
   ## Determine the number of words per line
   format = strrep (format, "%", " %");
-  [~, ~, ~, fmt_words] = regexp (format, "[^\\s]+");
+  [~, ~, ~, fmt_words] = regexp (format, "[^ ]+");
 
   num_words_per_line = numel (fmt_words);
   for m = 1:numel(fmt_words)
     ## Convert formats such as "%Ns" to "%s" (see the FIXME below)
     if (length (fmt_words{m}) > 2)
       if (strcmp (fmt_words{m}(1:2), "%*"))
         fmt_words{m} = "%*";
       elseif (fmt_words{m}(1) == "%")
